docs/DesignDocs/SealingAPI.md

```diff
@@ -0,0 +1,591 @@
+`additional_data_size`, respectively.
+
+```C
+uint8_t* blob;
+size_t blob_size;
+result = oe_seal(key_info, key_info_size,
+                 NULL,          // plaintext
+                 0,             // plaintext_size
+                 my_data,       // additional_data
+                 my_data_size,  // additional_data_size
+                 &blob, &blob_size);
+id (result != OE_OK)
+    // Handle encryption error...
+else
+{
+    // Persist blob...
+
+    oe_free(blob); // Free blob when done
+}
+```
+
+Please note that the blob format is proprietary so there's no architectural way
+to extract `additional_data` or `additional_data_size` without unsealing the
+blob. That said, the only way to recover `my_data` (and `my_data_size`) is
+still via `oe_unseal()`, regardless it was encrypted or not. In other words,
+the two code snippets above are identical from functional perspective but
+differ in security. Therefore, developers are strongly encouraged to use
+encryption for comprehensive protection.
+
+`oe_unseal()` decrypts a sealed blob **in-place**. The code snippet below shows
+how to decrypt a sealed blob and retrieve a pointer to the plaintext. Please
+note that `my_data` after this call will point to somewhere within `blob`.
+
+```C
+result = oe_unseal(blob, blob_size,
+                   &my_data,        // plaintext
+                   &my_data_size,   // plaintext_size
+                   NULL,            // additional_data
+                   NULL             // additional_data_size
+                   );
+if (result != OE_OK)
+    // Handle decryption error...
+else
+{
+    // Process my_data...
+}
+```
+
+The initial `key_info` passed to `oe_seal()` could be retrieved by a call to
+`oe_get_seal_key_by_policy()`, which takes a `seal_policy` as its only input
+parameter. In practice, an enclave may desire to encrypt different things under
+different keys. Per
+[NIST.SP800-108](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-108.pdf),
+*Label* and *Context* should be among the inputs to *KDF* (*Key Derivation
+Function*). Most TEEs also support taking addtional *context* or *entropy* into
+key derivation. The existing `oe_get_seal_key_by_policy()` API however lacks
+that support and prompts for a new API to fill the gap.
+
+The new `oe_get_seal_key_info()` API is introduced with the following
+signature.
+
+```C
+oe_result_t oe_get_seal_key_info(
+    oe_seal_policy_t seal_policy,
+    uint8_t* entropy,
+    size_t entropy_size,
+    uint64_t tee_specific,
+    uint8_t** key_info,
+    size_t* key_info_size);
+```
+
+If compared to `oe_get_seal_key_by_policy()`, `oe_get_seal_key_info()` differs
+in:
+
+- The addition of `entropy`/`entropy_size` parameters - This allows *Label* or
+  *Context* (or both), or simply more entropy to be mixed into the seal key.
+- The addition of `tee_specific` parameter - This is a TEE specific parameter,
+  or `OE_SEAL_SGX` to use TEE defaults. See
```

Comment:
Above line looks wrong.  OE_SEAL_SGX won't use TEE defaults on non-SGX TEEs.

Reply:
`OE_SEAL_SGX` won't be defined on non-SGX TEEs and will lead to a compiler error.